### PR TITLE
Build: Reports times and outcomes from CircleCI jobs

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
As a first attempt to start dogfooding Grafana this PR will enable collection of build times and build outcomes for all jobs in CircleCI for the `master` branch only.

[Dogfooding dashboard](https://play.grafana.org/d/4WevhMxWk/ci-times-and-outcomes?orgId=1)

There are things that could be improved in future PRs like:

1. Combining `ci-job-succeeded.sh` and `ci-job-failed.sh` into the same script and calling that with a flag instead (I'm not a bash expert so I could probably do this with some guidance)
2. Using CircleCI Orbs for reusing the steps? (???)
3. Reporting on WorkFlow level? (Maybe this can be done in Grafana instead?)

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by torkelo** [comment]
Nice! 

Searched a bit for CircleCI metrics apis or other ways we can collect this and found this:
https://circleci.com/orbs/registry/orb/circleci/sumologic 

If you go down to the section Orb Source. you can see how it is implemented. It is some workflow that you add to other other steps that runs in parallel and collects metrics. So it looks like you do not need to modify your orther workflows/jobs as much, just need to add this collector job/workflow.  

Not sure how easy that is to adapt but looks interesting.

**Comment by hugohaggmark** [comment]
> Nice!
> 
> Searched a bit for CircleCI metrics apis or other ways we can collect this and found this:
> https://circleci.com/orbs/registry/orb/circleci/sumologic
> 
> If you go down to the section Orb Source. you can see how it is implemented. It is some workflow that you add to other other steps that runs in parallel and collects metrics. So it looks like you do not need to modify your orther workflows/jobs as much, just need to add this collector job/workflow.
> 
> Not sure how easy that is to adapt but looks interesting.

I think `Orbs`would tie our CI pipeline even more to CircleCI and I'm not sure that is good

**Review by torkelo — COMMENTED** [review]
Yes, that is true. Just would be nice to centralize the ci, start, fail & succeed config and not have to add them to every job. Not sure if that is possible

**Comment by hugohaggmark** [comment]
> Yes, that is true. Just would be nice to centralize the ci, start, fail & succeed config and not have to add them to every job. Not sure if that is possible

My suggestion is to look into that in future PRs :)

File: .circleci/config.yml [hunk 1]
```diff
@@ -48,13 +59,24 @@ jobs:
     working_directory: /go/src/github.com/grafana/grafana
     steps:
         - checkout
+        - run:
+            name: ci job started
+            command: './scripts/ci-job-started.sh'
```
Comment (dprokop) [inline .circleci/config.yml]: Maybe we should only wrap the part of the jobs we have control over? `apt update`, ` restore-cache`, `yarn install` etc might cause the build to fail but this is sth we don't have control over. Sort of depends on what we want to see from these metrics.

**Comment by papagian** [comment]
> Yes, that is true. Just would be nice to centralize the ci, start, fail & succeed config and not have to add them to every job. Not sure if that is possible

Depending how much we intend to be tied to CircleCI but a solution to this would be to use [Steps](https://circleci.com/docs/2.0/reusing-config/#steps).
I that case we could define a `publish-stats` command like:
```version: 2.1
commands:
  publish-stats:
    parameters:
      after-start:
        description: "Steps that will be executed after ci-job-started, but ci-job-failed or ci-job-succeeded"
        type: steps
        default: []
    steps:
    - run:
        name: ci job started
        command: './scripts/ci-job-started.sh'
    - steps: << parameters.after-start >>
    - run:
        name: ci job failed
        command: './scripts/ci-job-failed.sh'
        when: on_fail
    - run:
        name: ci job succeeded
        command: './scripts/ci-job-succeeded.sh'
        when: on_success
```

and then we could define for instance the `mysql-integration-test` job as:


```
version: 2.1
jobs:
  mysql-integration-test:
    docker:
      - image: circleci/golang:1.13.1
      - image: circleci/mysql:5.6-ram
        environment:
          MYSQL_ROOT_PASSWORD: rootpass
          MYSQL_DATABASE: grafana_tests
          MYSQL_USER: grafana
          MYSQL_PASSWORD: password
    working_directory: /go/src/github.com/grafana/grafana
    steps:
        - checkout
        - publish-stats:
            after-start: 
                - run: sudo apt update
                - run: sudo apt install -y default-mysql-client
                - run: dockerize -wait tcp://127.0.0.1:3306 -timeout 120s
                - run: cat devenv/docker/blocks/mysql_tests/setup.sql | mysql -h 127.0.0.1 -P 3306 -u root -prootpass
                - run:
                    name: mysql integration tests
                    command: './scripts/circle-test-mysql.sh'
```

I haven't tested myself but I think that with small tweaks the above should work.
Also, if we want to use Steps directive we need to [upgrade](https://github.com/grafana/grafana/issues/18401) to CircleCI 2.1.

**Review by papagian — APPROVED** [review]
I have left a comment for a possible improvement otherwise LGTM!

**Comment by hugohaggmark** [comment]
> I have left a comment for a possible improvement otherwise LGTM!

Great suggestion @papagian will have a look today

**Comment by hugohaggmark** [comment]
> > I have left a comment for a possible improvement otherwise LGTM!
> 
> Great suggestion @papagian will have a look today

On second thought that should probably be another PR, not sure I want to upgrade CircleCI config in this one as well.

File: .circleci/config.yml (hunk 1 above)
Comment (hugohaggmark) [reply .circleci/config.yml]: I wanted to keep it simple so didn't think to much about this.

**Pull request merged**